src/com/jwetherell/algorithms/mathematics/Division.java

```diff
@@ -78,6 +99,11 @@ public static final long divisionUsingShift(int a, int b) {
     public static final long divisionUsingLogs(int a, int b) {
         long absA = Math.abs(a);
         long absB = Math.abs(b);
+
```

Comment:
The method will throw an exception by ensuring the value of b before the result "long" is calculated avoiding unexpected results.